pontoon/sync/tests/test_tasks.py

```diff
@@ -15,3 +15,47 @@ def test_missing_project(self):
             sync_project(99999)
             mock_log.error.assert_called_with(CONTAINS('99999'))
             assert_false(mock_perform_sync.called)
+
+class SyncExecutionTests(TestCase):
+    def setUp(self):
+        self.project = ProjectFactory.create()
+
+    def test_primary_process(self):
+        """
+        Test if sync will create lock in cache and release this after task is done.
+        """
+        with patch('django.core.cache.cache') as mock_cache, \
```

Comment:
Mocking the cache and ensuring it sets the right cache keys is a decent way to unit test this, but IMO it's brittle and we don't want to unit test this. The thing we care about is that we can't run two instances of this task at the same time. So what we really want to do is create a function that we pause during execution, and then attempt to run another function at the same time and observe the resulting error.

A reasonable approach would be to write a function decorated with the`@run_once_only` decorator that takes in a callback and executes it. Then, we call it and pass in a callback that attempts to call the decorated function again, and asserts that it raises an exception, since it's already in the middle of running once.

This also avoids testing if `sync_project` is decorated with the decorator, and only tests if the decorator does what we think it does.